lib/splitclient-rb/clients/split_client.rb

```diff
@@ -164,6 +178,33 @@ def sanitize_split_names(calling_method, split_names)
 
     private
 
+    def fix_properties(properties)
+      properties_count = 0
+      non_string_keys = false
+
+      fixed_properties = properties.each_with_object({}) { |(key, value), result|
```

Comment:
are you doing a loop for checking and then another for calculating the size? why don't add the logic on this `each` method instead of doing two loops?

Reply:
updated to a single loop